drake/util/drakeGradientUtil.h

```diff
@@ -30,6 +31,31 @@ std::array<int, Size> intRange(int start)
   return ret;
 }
 
+namespace Drake {
+  // todo: recursive template to get arbitrary gradient order
+
+  // note: tried using template default values (e.g. Eigen::Dynamic), but they didn't seem to work on my mac clang
+  template <int num_vars> using TaylorVar = Eigen::AutoDiffScalar< Eigen::Matrix<double,num_vars,1> >;
+  template <int num_vars, int cols> using TaylorVec = Eigen::Matrix< TaylorVar<num_vars>, cols, 1>;
+  template <int num_vars, int rows, int cols> using TaylorMat = Eigen::Matrix< TaylorVar<num_vars>, rows, cols>;
+
+  typedef TaylorVar<Eigen::Dynamic> TaylorVarX;
+  typedef TaylorVec<Eigen::Dynamic,Eigen::Dynamic> TaylorVecX;
+  typedef TaylorMat<Eigen::Dynamic,Eigen::Dynamic,Eigen::Dynamic> TaylorMatX;
+
+  // initializes the vector with x=val and dx=eye(numel(val))
+  template <typename EigenVec>
+  TaylorVecX initTaylorVecX(EigenVec val) {
```

Comment:
Why not follow the standard way of writing functions taking Eigen parameters (templating on Derived)? Also, shouldn't pass by value.